### PR TITLE
Setup publish pre-release to Bintray on merge into master

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

**Comment by olafurpg** [comment]
No need to wait for CI since these are only build changes. The interesting bits happen after merge into master.

**Comment by olafurpg** [comment]
Pleasing to see the publish succeeded on first try https://travis-ci.org/scalameta/scalameta/jobs/252315407#L5624

File: build.sbt [hunk 1]
```diff
@@ -552,10 +552,6 @@ lazy val publishableSettings = Def.settings(
   },
   sharedSettings,
   bintrayOrganization := Some("scalameta"),
-  bintrayRelease := (Def.taskDyn {
-    if (isCustomRepository) Def.task(())
-    else bintrayRelease
-  }).value,
```
Comment (xeno-by) [inline build.sbt]: Why was this removed?

Comment (olafurpg) [inline build.sbt]: See commit message https://github.com/scalameta/scalameta/pull/1016/commits/3b0c6cc3038c58391b401fc6d42b31c68ed262e4

Comment (olafurpg) [inline build.sbt]: Follow up here https://github.com/scalameta/scalameta/pull/1017/files
